fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/NamespaceTools.java

```diff
@@ -85,7 +85,7 @@ public static void validatePath(final Session session, final String path) {
             final int colonPosition = segment.indexOf(':');
             if (segment.length() > 0 && colonPosition > -1) {
                 final String prefix = segment.substring(0, colonPosition);
-                if (!prefix.equals("fedora")) {
+                if (!prefix.equals("fedora") && !prefix.equals("fcr")) {
```

Comment:
Why is the addition of the `fcr` prefix now being skipped?

Reply:
I think if it fails to retrieve the Fedora resource, it'll fall down to the validatePath(jcrSession, path) method above from https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/rdf/HttpResourceConverter.java#L148, which will have the 400 error for "Prefix has not been registered" instead of status 405 we want with PUTing, POSTing or PATCHing to a LDCPv/Memento path.

Reply:
Yeah this in a by-product of the regex being discussed above not matching on the a URI like `/fcrepo/rest/versioned_resource/fcr:versions/new`.

When it matches you get a PathNotFoundException before here, when it doesn't match it tries to find the `/fcr:versions`.

Alternatively we could duplicate the work in `FedoraLdp` for `GET` and move it to `FedoraVersioning` to match particularly on paths like `/fcr:versions/.*`. But that could be a lot of duplicated code.

Reply:
I would suggest a comment be added to explain why the `fcr` prefix is being disregarded. In that way, we can more easily know the implications of potentially removing the `!prefix.equals("fcr")` check in the future.

Reply:
Sure. I'll add a comment for it.